core/src/test/java/io/confluent/kafka/schemaregistry/rest/protobuf/RestApiTest.java

```diff
@@ -0,0 +1,207 @@
+    props.setProperty("schema.providers", ProtobufSchemaProvider.class.getName());
+    return props;
+  }
+
+  @Test
+  public void testBasic() throws Exception {
+    String subject1 = "testTopic1";
+    String subject2 = "testTopic2";
+    int schemasInSubject1 = 10;
+    List<Integer> allVersionsInSubject1 = new ArrayList<Integer>();
+    List<String> allSchemasInSubject1 = getRandomProtobufSchemas(schemasInSubject1);
+    int schemasInSubject2 = 5;
+    List<Integer> allVersionsInSubject2 = new ArrayList<Integer>();
+    List<String> allSchemasInSubject2 = getRandomProtobufSchemas(schemasInSubject2);
+    List<String> allSubjects = new ArrayList<String>();
+
+    // test getAllSubjects with no existing data
+    assertEquals("Getting all subjects should return empty",
+        allSubjects,
+        restApp.restClient.getAllSubjects()
+    );
+
+    // test registering and verifying new schemas in subject1
+    int schemaIdCounter = 1;
+    for (int i = 0; i < schemasInSubject1; i++) {
+      String schema = allSchemasInSubject1.get(i);
+      int expectedVersion = i + 1;
+      registerAndVerifySchema(restApp.restClient, schema, schemaIdCounter, subject1);
+      schemaIdCounter++;
+      allVersionsInSubject1.add(expectedVersion);
+    }
+    allSubjects.add(subject1);
+
+    // test re-registering existing schemas
+    for (int i = 0; i < schemasInSubject1; i++) {
+      int expectedId = i + 1;
+      String schemaString = allSchemasInSubject1.get(i);
+      int foundId = restApp.restClient.registerSchema(schemaString,
+          ProtobufSchema.TYPE,
+          Collections.emptyList(),
+          subject1
+      );
+      assertEquals("Re-registering an existing schema should return the existing version",
+          expectedId,
+          foundId
+      );
+    }
+
+    // test registering schemas in subject2
+    for (int i = 0; i < schemasInSubject2; i++) {
+      String schema = allSchemasInSubject2.get(i);
+      int expectedVersion = i + 1;
+      registerAndVerifySchema(restApp.restClient, schema, schemaIdCounter, subject2);
+      schemaIdCounter++;
+      allVersionsInSubject2.add(expectedVersion);
+    }
+    allSubjects.add(subject2);
+
+    // test getAllVersions with existing data
+    assertEquals(
+        "Getting all versions from subject1 should match all registered versions",
+        allVersionsInSubject1,
+        restApp.restClient.getAllVersions(subject1)
+    );
+    assertEquals(
+        "Getting all versions from subject2 should match all registered versions",
+        allVersionsInSubject2,
+        restApp.restClient.getAllVersions(subject2)
+    );
+
+    // test getAllSubjects with existing data
+    assertEquals("Getting all subjects should match all registered subjects",
+        allSubjects,
+        restApp.restClient.getAllSubjects()
+    );
+  }
+
+  @Test
+  public void testSchemaReferences() throws Exception {
+    Map<String, String> schemas = getProtobufSchemaWithReferences();
```

Comment:
what do you think about using the word `dependencies` instead of `references` here and in some other places? this seems more unambiguous to me. 

I think the name `SchemaReference` shouldn't change.

Reply:
Ack